src/parser/lexer.cpp

```diff
@@ -1069,6 +1069,57 @@ bool Lexer::takeRParen() {
   return false;
 }
 
+std::optional<std::string_view> Lexer::takeKeyword() {
+  if (curr) {
+    return std::nullopt;
```

Comment:
Can you explain this part? It reads as "if there is a current token, ignore it and return nullopt". Why is that correct?

Reply:
This goes away at the end of the patch sequence, but it is required for correctness at these intermediate steps. Let's say the lexer reaches an integer and eagerly lexes it into a token stored in `curr`. That integer may be followed by a keyword, but `takeKeyword()` should fail until `takeU32()` (or similar) has been called first. Without this check, `takeKeyword()` would succeed even though the preceeding integer was never consumed.

Reply:
I see, thanks. Which is the PR where I can see the final step where it goes away, to get an idea for the direction?

Reply:
#6544